public/src/components/switchboard.tsx

```diff
@@ -97,61 +136,164 @@ const Switchboard: React.FC<InnerProps<SupportFrontendSwitches>> = ({
   const classes = useStyles();
 
   const [needToSaveDataWarning, setNeedToSaveDataWarning] = useState(false);
+  const [newUnsavedData, setNewUnsavedData] = useState(false);
```

Comment:
`setPendingChanges` and `setNeedToSaveDataWarning` are called everywhere `setNewUnsavedData` is called. Could we get rid of these two `useStates`, and check if there are any pending changes instead? (I.e. on line 150 below we could replace `needToSaveDataWarning && newUnsavedData &&` with `(pendingChanges.length > 0) &&`.)